Firebase/Core/FIRLogger.m

```diff
@@ -242,18 +352,26 @@ void FIRLogBasic(FIRLoggerLevel level,
       stringWithFormat:@"%s - %@[%@] %@", FirebaseVersionString, service, messageCode, logMsg];
   dispatch_async(sFIRClientQueue, ^{
     asl_log(sFIRLoggerClient, NULL, level, "%s", logMsg.UTF8String);
-  });
 
-  // Keep count of how many errors and warnings are triggered.
-  if (level == FIRLoggerLevelError) {
-    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
-    NSInteger errorCount = [defaults integerForKey:kFIRLoggerErrorCountKey];
-    [defaults setInteger:errorCount + 1 forKey:kFIRLoggerErrorCountKey];
-  } else if (level == FIRLoggerLevelWarning) {
-    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
-    NSInteger warningCount = [defaults integerForKey:kFIRLoggerWarningCountKey];
-    [defaults setInteger:warningCount + 1 forKey:kFIRLoggerWarningCountKey];
-  }
+    // Keep count of how many errors and warnings are triggered.
+    if (level == FIRLoggerLevelError) {
```

Comment:
Can we do merge the logic here?
NSString *path;
if (level == FIRLoggerLevelError) { 
  path = FIRLoggerPathInCachesByAppending(kFIRLoggerErrorCountFileName);
} else if (level == FIRLoggerLevelWarning) {
  path = FIRLoggerPathInCachesByAppending(kFIRLoggerWarningCountFileName);
}
NSInteger currentValue = FIRReadIntegerFromFile(path);
if (currentValue <= kFIRLoggerCountFileNotFound) {
    FIRWriteIntegerToFile(1, path);
} else {
     FIRWriteIntegerToFile(currentValue + 1, path);
}

Reply:
Great idea, done!